### PR TITLE
Feature/getBlock

### PR DESCRIPTION
**Pull request by bzuker**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/blockcoders/near-rpc-providers/blob/main/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/near-rpc-provider.ts [hunk 1]
```diff
@@ -154,10 +162,75 @@ export class NearRpcProvider extends JsonRpcProvider {
...
+    const blockResponse = await this.send<BlockRpcResponse>('block', { block_id: params.block_id })
+
+    return {
+      hash: blockResponse.header.hash,
+      parentHash: blockResponse.header.prev_hash,
+      number: blockResponse.header.height,
+      timestamp: blockResponse.header.timestamp,
+      // TODO: starting from here, are these values correct?
+      nonce: blockResponse.header.random_value,
+      difficulty: blockResponse.header.block_ordinal,
```
Comment (jarcodallo) [inline src/near-rpc-provider.ts:200]: The block ordinal is the amount of Blocks already existing on the Canonical Chain + 1

File: src/near-rpc-provider.spec.ts [hunk 2]
```diff
@@ -26,7 +26,26 @@ describe('NearRpcProvider', () => {
...
+      const block = await provider.getBlock(74419929)
+      expect(block).to.not.be.undefined
+    })
+
+    it('should get the block by hash', async () => {
+      const block = await provider.getBlock('DckezGRnns3cYBwj1A9KPajpu3pnabUGE3bedJEgJnSU')
+      expect(block).to.haveOwnProperty('hash')
+    })
+
+    it('should get the same block by hash and number', async () => {
```
Comment (jarcodallo) [inline src/near-rpc-provider.spec.ts:43]: love it ❤️

**Review by AnnRiera — APPROVED** [review]
LGTM!

File: src/near-rpc-provider.spec.ts [hunk 3]
```diff
@@ -26,7 +26,26 @@ describe('NearRpcProvider', () => {
     it('should get the balance for the account', async () => {
       const balance = await provider.getBalance('blockcoders.testnet')
       expect(balance).to.be.instanceOf(BigNumber)
-      expect(balance.gt(BigNumber.from(0))).to.be.true
+    })
+  })
+
+  describe('getBlock', () => {
```
Comment (jarcodallo) [inline src/near-rpc-provider.spec.ts:32]: we have to add a test when you pass null or undefined. If you do, that should return the latest block.

**Review by jarcodallo — APPROVED** [review]
LGTM, please add a test for the null/undefined case.

File: src/near-rpc-provider.spec.ts (hunk 3 above)
Comment (bzuker) [reply src/near-rpc-provider.spec.ts:32]: as discussed, not possible with the current ethers interface
